### PR TITLE
port firebase lib to api v3

### PR DESCRIPTION
**Pull request by paglias**

### CODE REVIEW

File: website/src/libs/api-v3/firebase.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import Firebase from 'firebase';
+import nconf from 'nconf';
+const IS_PROD = nconf.get('IS_PROD');
+const FIREBASE_CONFIG = nconf.get('FIREBASE');
+const FIREBASE_ENABLED = IS_PROD && FIREBASE_CONFIG.ENABLED === 'true';
```
Comment (crookedneighbor) [inline website/src/libs/api-v3/firebase.js:5]: Do we want to make this a prod only thing? Can devs sign up for test accounts with Firebase?

File: website/src/libs/api-v3/firebase.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+let firebaseRef;
```
Comment (crookedneighbor) [inline website/src/libs/api-v3/firebase.js:7]: I think this should just be 'firebase'. That's the normal convention for declaring instances of an object.

File: website/src/models/group.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ var _ = require('lodash');
 var async = require('async');
 var logging = require('../libs/api-v2/logging');
 var Challenge = require('./../models/challenge').model;
-var firebase = require('../libs/firebase');
+var firebase = require('../libs/api-v2/firebase');
```
Comment (crookedneighbor) [inline website/src/models/group.js:9]: Should this be v3?

**Comment by crookedneighbor** [comment]
There is no actual content difference between 2 and 3, right? Should the models just reference v3 then?

**Comment by crookedneighbor** [comment]
Looks good. Just had a few minor comments.

File: website/src/libs/api-v3/firebase.js (hunk 2 above)
Comment (paglias) [inline website/src/libs/api-v3/firebase.js:7]: yeah but in this case it's not a normal instance, but a reference to a specific path in firebase

File: website/src/models/group.js (hunk 3 above)
Comment (paglias) [inline website/src/models/group.js:9]: I plan to change this when we migrate the groups model

**Comment by paglias** [comment]
Yeah no difference, I'm going to delete the v2 version later, we'll also delete the logging library and a few other things

File: website/src/libs/api-v3/firebase.js (hunk 1 above)
Comment (paglias) [inline website/src/libs/api-v3/firebase.js:5]: Yes, you can signup for a dev account

Comment (crookedneighbor) [inline website/src/libs/api-v3/firebase.js:5]: I think the check should just be for FIREBASE_ENABLED.

Comment (paglias) [inline website/src/libs/api-v3/firebase.js:5]: ok, changing it

**Pull request merged**